modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/operation/ValueReturnDelegateContractTestCase.java

```diff
@@ -203,6 +210,28 @@ public void operationWithResultInputStreamOutput() throws Exception {
     assertStreamIsWrapped(Result.builder().output(new ByteArrayInputStream(HELLO_WORLD_MSG.getBytes(UTF_8))).build());
   }
 
+  @Test
+  public void operationWithResultInputStreamCollectionOutput() throws Exception {
+    MetadataType metadataType = BaseTypeBuilder.create(JAVA).arrayType().of(new MessageMetadataTypeBuilder().build()).build();
+    when(outputModel.getType()).thenReturn(metadataType);
+
+    delegate = createReturnDelegate();
+
+    List<Result> resultList = new ArrayList<>();
+    resultList.add(Result.builder().output(new ByteArrayInputStream(HELLO_WORLD_MSG.getBytes(UTF_8))).build());
+
+    CoreEvent result = delegate.asReturnValue(resultList, operationContext);
+
+    Message message = getOutputMessage(result);
+    Message message1 = ((ResultsToMessageList) message.getPayload().getValue()).get(0);
+
+    ManagedCursorStreamProvider actual = (ManagedCursorStreamProvider) message1.getPayload().getValue();
+    InputStream resultingStream = actual.openCursor();
+    assertThat(IOUtils.toString(resultingStream), is(HELLO_WORLD_MSG));
+    resultingStream.close();
+    actual.releaseResources();
+  }
```

Comment:
In the case of files there shouldn't exists a _connectionHandler_.  Indeed, debugging the code I verified that _CONNECTION_PARAM_ is always null when using the file connector.

I can included it in the test since it's mocked, so should be called. But I wasn't testing that.